core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -0,0 +1,221 @@
+    for (endOfAuthority in endsOfAuthority) {
+        val targetPosition = endOfAuthority.offsetEOA.distance.meters
+        val targetSpeed = 0.0
+        val overhead =
+            Envelope.make(
+                EnvelopePart.generateTimes(
+                    listOf(EnvelopeProfile.CONSTANT_SPEED),
+                    doubleArrayOf(0.0, targetPosition),
+                    doubleArrayOf(envelope.maxSpeed, envelope.maxSpeed)
+                )
+            )
+        val guiCurve =
+            computeBrakingCurve(
+                context,
+                overhead,
+                targetPosition,
+                targetSpeed,
+                BrakingType.ETCS_GUI
+            )
+        val sbdCurve =
+            computeBrakingCurve(
+                context,
+                overhead,
+                targetPosition,
+                targetSpeed,
+                BrakingType.ETCS_SBD
+            )
+        val fullIndicationCurve =
+            computeIndicationBrakingCurveFromRef(context, sbdCurve, BrakingCurveType.SBD, guiCurve)
+        val indicationCurve = keepBrakingCurveUnderOverlay(fullIndicationCurve, envelope)
+        builder.addPart(indicationCurve)
+    }
+    return builder.build()
+}
+
+/** Compute braking curve: used to compute EBD, SBD or GUI. */
+private fun computeBrakingCurve(
+    context: EnvelopeSimContext,
+    envelope: Envelope,
+    targetPosition: Double,
+    targetSpeed: Double,
+    brakingType: BrakingType
+): EnvelopePart {
+    // If the stopPosition is below zero, the input is invalid
+    // If the stopPosition is after the end of the path, the input is invalid except if it is an
+    // SVL, i.e. the target speed is 0 and the curve to compute is an EBD
+    if (
+        targetPosition <= 0.0 ||
+            (targetPosition > context.path.length &&
+                targetSpeed == 0.0 &&
+                brakingType != BrakingType.ETCS_EBD)
+    )
+        throw RuntimeException(
+            String.format(
+                "Trying to compute ETCS braking curve from out of bounds ERTMS end/limit of authority: %s",
+                targetPosition
+            )
+        )
+    val partBuilder = EnvelopePartBuilder()
+    partBuilder.setAttr(EnvelopeProfile.BRAKING)
+    val overlayBuilder =
+        ConstrainedEnvelopePartBuilder(
+            partBuilder,
+            SpeedConstraint(0.0, EnvelopePartConstraintType.FLOOR),
+            EnvelopeConstraint(envelope, EnvelopePartConstraintType.CEILING)
+        )
+    EnvelopeDeceleration.decelerate(
+        context,
+        targetPosition,
+        targetSpeed,
+        overlayBuilder,
+        -1.0,
+        brakingType
+    )
+    val brakingCurve = partBuilder.build()
+    return brakingCurve
+}
+
+/** Compute Indication curve: EBI/SBD -> SBI -> PS -> IND */
+private fun computeIndicationBrakingCurveFromRef(
+    context: EnvelopeSimContext,
+    refCurve: EnvelopePart,
+    refBrakingCurveType: BrakingCurveType,
```

Comment:
For consistency:
```suggestion
    refBrakingCurve: EnvelopePart,
    refBrakingCurveType: BrakingCurveType,
```
(no `Braking` at all also works for me)

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/10014/commits/c1bc4feb0ae2e7792d1eb8b8c9171b225d546229